src/player.cpp

```diff
@@ -9610,32 +9610,41 @@ bool player::wield( item& target )
 
     int mv = 0;
 
+    bool target_removed = false;
+    item target_copy = target;
     if( is_armed() ) {
+        inventory old_inv = inv;
```

Comment:
> would it still look hack-ish?

If you can avoid unnecessary copying then no, it won't. Otherwise it would. Volume can be estimated, but please remember that `volume_capacity()` isn't linear since it uses multipliers for the `PACKMULE` and `DISORGANIZED` traits.